accounts/serializer.py

```diff
@@ -0,0 +1,277 @@
+from rest_framework_simplejwt.tokens import RefreshToken,TokenError
+from django.contrib import auth
+class UserProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Account
+        fields = ( 'name','display_picture','bio','created_at','updated_at')
+
+
+class UserSerializer(serializers.HyperlinkedModelSerializer):
+    """
+    Bifrost user writable nested serializer
+    """
+    profile = UserProfileSerializer(required=True)
+
+    class Meta:
+        model = User
+        fields = ('url', 'email','username', 'first_name', 'last_name', 'password', 'profile','verification_code','phone_number',)
+        extra_kwargs = {'password': {'write_only': True}}
+     
+
+    def create(self, validated_data):
+        profile_data = validated_data.pop('profile')
+        password = validated_data.pop('password')
+        user = User(**validated_data)
+        user.set_password(password)
+        user.save()
+        Account.objects.create(user=user, **profile_data)
+        return user
+
+    def update(self, instance, validated_data):
+        profile_data = validated_data.pop('profile')
+        profile = instance.profile
+
+        instance.email = validated_data.get('email', instance.email)
+        instance.save()
+
+        
+        profile.name = profile_data.get('name', profile.name)
+        profile.display_picture = profile_data.get('display_picture', profile.display_picture)
+        profile.bio = profile_data.get('bio', profile.bio)
+        profile.created_at = profile_data.get('created_at', profile.created_at)
+        profile.updated_at = profile_data.get('updated_at', profile.updated_at)
+       
+        profile.save()
+
+        return instance
+
+
+class LogoutSerializer(serializers.Serializer):
+    refresh=serializers.CharField()
+    default_error_messages={
+        'bad_token':('Token is expired or invalid')
+    }
+    def validate(self, attrs):
+        self.token = attrs['refresh']
+
+        return attrs
+
+    def save(self, **kwargs):
+        try:
+            RefreshToken(self.token).blacklist()
+        except TokenError:
+            self.fail('bad_token')
+        
+class LoginSerializer(serializers.ModelSerializer):
+    email=serializers.EmailField(max_length=255,min_length=3)
+    password=serializers.CharField(max_length=68, min_length=6,write_only=True)
+    username=serializers.CharField(max_length=255,min_length=3, read_only=True)
+    
+    tokens=serializers.SerializerMethodField()
+
+    def get_tokens(self,obj):
+        user = User.objects.get(email=obj['email'])
+
+        return {
+            'access':user.tokens()['access'],
+            'refresh':user.tokens()['refresh']
+        }
+    
+    class Meta:
+        model=User
+        fields=['email','password','username','tokens']
+
+    def validate(self, attrs):
+        email = attrs.get('email', '')
+        password=attrs.get('password', '')
+
+        user = auth.authenticate(email=email, password=password)
```

Comment:
I see this is where the user gets authenticated and you are using **rest_framework_simplejwt** to do all the heavy lifting for you. From the docs, is **rest_framework_simplejwt** storing the refresh token as a [**http_only**](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#security) cookie ? Or you just send the access and refresh tokens as a response object directly to the user ?